src/app/my-tab/my-tab.component.ts

```diff
@@ -0,0 +1,140 @@
+import {
+  Component,
+  OnInit,
+  OnDestroy
+} from '@angular/core';
+
+import {
+  AddinClientService
+} from '@blackbaud/skyux-lib-addin-client';
+
+import {
+  AddinClientInitArgs
+} from '@blackbaud/sky-addin-client';
+
+import {
+  Subject
+} from 'rxjs';
+
+import {
+  takeUntil,
+  finalize
+} from 'rxjs/operators';
+
+import {
+  GiftFormData
+} from '../shared/interfaces/gift-form-data';
+
+import {
+  GiftSaveData
+} from '../shared/interfaces/gift-save-data';
+
+import {
+  EntryFormService
+} from '../shared/services/entry-form.service';
+
+@Component({
+  selector: 'my-tab',
+  templateUrl: './my-tab.component.html',
+  styleUrls: ['./my-tab.component.scss']
+})
+export class MyTabComponent implements OnInit, OnDestroy {
+  public environmentId: string;
+  public formData: GiftFormData;
+  public saveData: GiftSaveData;
+  private destroy: Subject<any>;
+
+  constructor(
+    private addinClientService: AddinClientService,
+    private entryFormService: EntryFormService
+  ) {
+    this.destroy = new Subject();
+  }
+
+  public ngOnDestroy() {
+    this.destroy.next();
+    this.destroy.complete();
+  }
+
+  public ngOnInit() {
+    // initialization of the add in client
+    this.initializeAddInClient();
+
+    // capture the update events triggered from the gift form through the add in client
+    this.captureGiftFormUpdateEvent();
+
+    // capture the cancel event triggered from the gift form through the add in client
+    this.captureGiftFormCancelEvent();
+
+    // capture the save event triggered from the gift form through the add in client
+    this.captureGiftFormSaveEvent();
```

Comment:
Maybe rename these to "handle...Event" rather than "capture...Event".  Handle is the naming we use in the API.

Reply:
Making this change now

Reply:
Renamed methods and comments to use 'handle' and not 'capture'